<launch>
    <arg name="use_rviz" default="true" doc="Controls whether rviz is launched-- default true"/>

    <node pkg="turtlebot3_teleop" name="turtlebot3_teleop_keyboard" type="turtlebot3_teleop_key" output="screen"/>
    
    <node pkg="nuturtlesim" type="tube_world" name="tube_world" output="screen" />

    <node name="rviz" pkg="rviz" args="-d $(find nuturtlesim)/config/model.rviz -f world" type="rviz"/>

    <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml"/>
    <rosparam command="load" file="$(find nuturtlesim)/config/tube_world_params.yaml"/>
    <rosparam command="load" file="$(find nuturtlesim)/config/frame_params.yaml"/>
    <rosparam command="load" file="$(find nuturtlesim)/config/scan_params.yaml"/>
</launch>